### PR TITLE
ENG-4653: Allowed dash character on page code and friendly-code

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

File: engine/src/main/java/org/entando/entando/web/page/model/PageRequest.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class PageRequest {
 
     @Size(max = 30, message = "string.size.invalid")
     @NotNull(message = "page.code.notBlank")
-    @Pattern(regexp = "[a-zA-Z0-9_-]+", message="page.code.wrongCharacters")
+    @Pattern(regexp = "[a-zA-Z0-9-_]+", message="page.code.wrongCharacters")
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/web/page/model/PageRequest.java]: is it possible to use a constant? the previous file has the same regexp ...

File: admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageAction.java [hunk 2]
```diff
@@ -49,7 +49,6 @@
 import java.util.Date;
 import java.util.Set;
 import org.entando.entando.ent.exception.EntException;
-import org.junit.jupiter.api.Assertions;
```
Comment (entgigi) [inline admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageAction.java:52]: this file has the wrong name, `Test` must be the last word in a filename, i remember some exclusions but it should be possible to find a name that matches our rule ...

File: engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java [hunk 3]
```diff
@@ -931,7 +931,86 @@ private PageRequest createPageRequest(String pageCode, String groupCode, String
...
+                            createPageRequest(
+                                    wrongPageCode,
+                                    Group.FREE_GROUP_NAME,
+                                    defaultParentCode))))
+                    .andDo(resultPrint());
+            result.andExpect(status().isBadRequest())
+                    .andExpect(jsonPath("$.payload.size()", is(0)))
+                    .andExpect(jsonPath("$.errors.size()", is(1)))
+                    .andExpect(jsonPath("$.errors[0].code", is("58")));
+            /*
```
Comment (entgigi) [inline engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java]: you should remove commented code or add comments to explain why

File: portal-ui/src/main/java/com/agiletec/aps/system/services/controller/control/RequestValidator.java [hunk 4]
```diff
@@ -198,20 +198,23 @@ protected String getNotFoundPageCode() {
...
     public void setLangManager(ILangManager langManager) {
-        this._langManager = langManager;
+        this.langManager = langManager;
     }
 
-    private ILangManager _langManager;
+    private transient ILangManager langManager;
 
+    /**
+     * @deprecated pattern for old suffix ".wp"
```
Comment (entgigi) [inline portal-ui/src/main/java/com/agiletec/aps/system/services/controller/control/RequestValidator.java]: you should add something like "you should use the new `pattern` variable", when you deprecate you should always explain the new solution, not only deprecation reason

File: engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java [hunk 5]
```diff
@@ -931,7 +931,86 @@ private PageRequest createPageRequest(String pageCode, String groupCode, String
         pageRequest.setParentCode(parentCode);
         return pageRequest;
     }
-
+    
+    @Test
+    void testValidatePage() throws Throwable {
```
Comment (entgigi) [inline engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java]: I think that the test name is not meaningful, can you try something like "shouldValidatorReturnServerErrorOverNullParameterTitlePageOwnerModel" ... if you agree.

File: portal-ui/src/test/java/org/entando/entando/aps/system/services/controller/control/TestRequestValidator.java [hunk 6]
```diff
@@ -27,6 +27,12 @@
 import com.agiletec.aps.system.services.lang.Lang;
 import com.agiletec.aps.system.services.page.IPage;
 import com.agiletec.aps.system.services.page.IPageManager;
+import com.agiletec.aps.system.services.page.Page;
+import com.agiletec.aps.system.services.page.PageMetadata;
+import com.agiletec.aps.system.services.page.PageTestUtil;
+import com.agiletec.aps.system.services.page.Widget;
+import com.agiletec.aps.system.services.pagemodel.IPageModelManager;
+import com.agiletec.aps.system.services.pagemodel.PageModel;
```
Comment (entgigi) [inline portal-ui/src/test/java/org/entando/entando/aps/system/services/controller/control/TestRequestValidator.java:35]: "Test" word should be at the end of filename ...

File: seo-plugin/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/controller/control/RequestValidator.java [hunk 7]
```diff
@@ -113,9 +113,9 @@ protected IContentManager getContentManager() {
     public void setContentManager(IContentManager contentManager) {
         this._contentManager = contentManager;
     }
-
-    protected Pattern patternSeoPath = Pattern.compile("^/page/(\\w+)((/\\w+)*+)");
-
+
```
Comment (entgigi) [inline seo-plugin/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/controller/control/RequestValidator.java:116]: regexps are hard to read, maybe adding a comment with the covered new  cases can be a good help for the future reader

**Review by entgigi — CHANGES_REQUESTED** [review]

**Comment by entgigi** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=entando_app-engine&pullRequest=117)
> 
> [Bug](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=entando_app-engine&pullRequest=117&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=entando_app-engine&pullRequest=117&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=entando_app-engine&pullRequest=117&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=CODE_SMELL) [B](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=CODE_SMELL) [3 Code Smells](https://sonarcloud.io/project/issues?id=entando_app-engine&pullRequest=117&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=entando_app-engine&pullRequest=117&metric=new_coverage&view=list) [100.0% Coverage](https://sonarcloud.io/component_measures?id=entando_app-engine&pullRequest=117&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=entando_app-engine&pullRequest=117&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=entando_app-engine&pullRequest=117&metric=new_duplicated_lines_density&view=list)

@eugeniosant  I think you can ignore this 3 smell code, put a comment, and go ahead with the process, just when you merge put a comment to explain why you ignored the sonar complaints (it's not the regexp the deprecation object, it's the use of .wp suffix ...)

File: portal-ui/src/main/java/com/agiletec/aps/system/services/controller/control/RequestValidator.java (hunk 4 above)
Comment (eugeniosant) [reply portal-ui/src/main/java/com/agiletec/aps/system/services/controller/control/RequestValidator.java]: This pattern in maintained to ensure compatibility with links used in older installations. Explained in javadoc

File: engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java]: I'll add a comment do explain that this test is regarding a missing validation. Added a new issue in internal board

File: admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageAction.java (hunk 2 above)
Comment (eugeniosant) [reply admin-console/src/test/java/com/agiletec/apsadmin/portal/TestPageAction.java:52]: Maybe it's better to create a ticket to fix the naming convention in whole source code

File: seo-plugin/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/controller/control/RequestValidator.java (hunk 7 above)
Comment (eugeniosant) [reply seo-plugin/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/controller/control/RequestValidator.java:116]: The new use case is the target of this pr; in this regex the target is to allow dash character into the friendly-code

File: engine/src/main/java/org/entando/entando/web/page/model/PageRequest.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/web/page/model/PageRequest.java]: Done

File: engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/test/java/org/entando/entando/web/page/PageControllerIntegrationTest.java]: Method renamed and divided in several tests

**Review by entgigi — APPROVED** [review]
LGTN

**Pull request merged**